### PR TITLE
Fix remote repo panic

### LINKED ISSUE #84: [BUG] panic: runtime error: invalid memory address or nil pointer dereference v0.3.0
Hi, thanks for that amazing tool. 

I found that version `v0.3.0` doesn't work while `v0.2.2` does work: 

```shell
docker run --rm -it -p 9000:9000 -v $(pwd):/src --env=AWS_ACCESS_KEY_ID="nanana" --env=AWS_SECRET_ACCESS_KEY="nununu" --env=AWS_DEFAULT_REGION="eu-west-1" --platform linux/amd64 im2nguyen/rover -tfBackendConfig lalala.s3.tfbackend -tfVarsFile lalala.tfvars -workspaceName=lalala
2022/01/14 11:59:40 Starting Rover...
2022/01/14 11:59:40 Initializing Terraform...
2022/01/14 11:59:58 Running in mkyc workspace...
2022/01/14 12:00:03 Generating plan...
2022/01/14 12:00:34 Parsing configuration...
2022/01/14 12:00:34 Generating resource overview...
2022/01/14 12:00:35 Generating resource map...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0xa5b0cf]

goroutine 1 [running]:
main.(*rover).GenerateModuleMap(0xc000159320, 0xc0003c87e0, {0xc00003d8d8, 0x12})
        /src/map.go:212 +0x1acf
main.(*rover).GenerateModuleMap(0xc000159320, 0xc0006ddb88, {0x0, 0x0})
        /src/map.go:240 +0xefc
main.(*rover).GenerateMap(0xc000159320)
        /src/map.go:319 +0x1bc
main.(*rover).generateAssets(0xc000159320)
        /src/main.go:207 +0x176
main.main()
        /src/main.go:142 +0xd19

```

when running the same with `v0.2.2` it works.

### PR DESCRIPTION
**Pull request by JackFlukinger**
Hey @im2nguyen, this should resolve #84 . It just checks to make sure the module is not at a remote url before trying to load the module (for line number and file info).

### CODE REVIEW

**Comment by JackFlukinger** [comment]
Also, I removed the ecs-cluster example as I don't think the submodule was actually pointing to anything. If u want to re-add it, definitely do. Also, I added a nested-module example just for testing. It actually points back at this repo and uses the `random-name` module, but it won't work until you push these changes to main.

**Review by im2nguyen — APPROVED** [review]
Thanks!

It's interesting how this doesn't work on remote modules since rover uses the terraform plan output. The init command downloads remote modules and stores it in the .terraform dir, when you run a plan or an apply, it _should_ use module in .terraform.

Even though this looks on a spot check, gonna verify the above before merging. (Aiming for Wed)

If I can't get to it by Wed, I'll just merge this as a stop gap 😄

**Comment by JackFlukinger** [comment]
Oh yeah @im2nguyen , line numbers and file names require `tfconfig.LoadModule(dir)`, which can only load modules in the filesystem. I think we can probably remove the -tfConfigExists flag (after testing ofc), I think configuration existence is checked for automatically at this point.

To access line numbers and file names of nested modules, it essentially calls `tfconfig.LoadModule(${parent-module-path}/${child-module-path})`. Of course, with remote modules the source is usually something like `git::https://github.com/im2nguyen/rover.git//example/random-test/random-name` which can't be loaded.

I'll see if there's a way to parse the module call and access the module in the .terraform folder. If so, I'll commit it to this PR :).

**Comment by JackFlukinger** [comment]
Okay @im2nguyen , it feels a bit hackish but these commits directly parse the `modules.json` file in the `.terraform` folder. This file contains local module paths with which we use to load them for file names and line numbers. 

I also removed the -tfConfigExists flag, since there's no longer any use for it. If the modules aren't available locally, rover just prints either `Continuing without loading module from filesystem`. If the root module isn't available, rover prints 		

`Could not load configuration from: ${workingDir}
Continuing without configuration file data...` 

Let me know if this seems like the right approach!

File: example/nested-test/main.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module "sub_module" {
+  source               = "./nested-module"
+
```
Comment (im2nguyen) [inline example/nested-test/main.tf:3]:
```suggestion
```

File: example/nested-test/nested-module/main.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module "remote_module" {
+  source = "git::https://github.com/im2nguyen/rover.git//example/random-test/random-name"
+
+  max_length = "3"
+
```
Comment (im2nguyen) [inline example/nested-test/nested-module/main.tf:5]:
```suggestion
```

**Review by im2nguyen — APPROVED** [review]
This looks good! I ran into the following error while testing on the nested-test directory, but I assumed this was bc the main branch still references `eks-cluster`. It worked when I tested using another remote module though! 😄 

```
$ ./rover -workingDir example/nested-test/
2022/01/24 10:33:45 Starting Rover...
2022/01/24 10:33:45 Initializing Terraform...
2022/01/24 10:33:55 Unable to parse Plan: Unable to initialize Terraform Plan: exit status 1

Error: Failed to download module

Could not download module "remote_module" (nested-module/main.tf:1) source
code from "git::https://github.com/im2nguyen/rover.git": error downloading
'https://github.com/im2nguyen/rover.git': /usr/local/bin/git exited with 128:
fatal: No url found for submodule path 'example/eks-cluster' in .gitmodules



Error: Failed to download module

Could not download module "remote_module" (nested-module/main.tf:1) source
code from "git::https://github.com/im2nguyen/rover.git": error downloading
'https://github.com/im2nguyen/rover.git': /usr/local/bin/git exited with 128:
fatal: No url found for submodule path 'example/eks-cluster' in .gitmodules
```

**Pull request merged**